pkg/agent/cniserver/ipam/antrea_ipam.go

```diff
@@ -272,9 +278,19 @@ func (d *AntreaIPAM) secondaryNetworkAdd(args *invoke.Args, k8sArgs *types.K8sAr
 	return result, nil
 }
 
+// SecondaryNetworkRelease releases IP addresses allocated for a Pod secondary network interface.
+func (d *AntreaIPAM) SecondaryNetworkRelease(owner *crdv1a2.PodOwner) error {
+	_, err := d.del(owner)
+	return err
+}
+
+func (d *AntreaIPAM) secondaryNetworkAdd(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig *types.NetworkConfig) (*current.Result, error) {
+	return d.SecondaryNetworkAllocate(getAllocationPodOwner(args, k8sArgs, nil, true), networkConfig)
+}
+
 func (d *AntreaIPAM) secondaryNetworkDel(args *invoke.Args, k8sArgs *types.K8sArgs, networkConfig *types.NetworkConfig) error {
-	owner := getAllocationOwner(args, k8sArgs, nil, true)
-	_, err := d.del(owner.Pod)
+	podOwner := getAllocationPodOwner(args, k8sArgs, nil, true)
+	_, err := d.del(podOwner)
```

Comment:
should this be `d.SecondaryNetworkRelease(getAllocationPodOwner(args, k8sArgs, nil, true))` for symmetry with `secondaryNetworkAdd`?

BTW, it's not super clear to me why we are even keeping `secondaryNetworkAdd` / `secondaryNetworkDel` / `secondaryNetworkCheck`? For example, it feels a bit strange to me to have `SecondaryNetworkAdd` (in ipam_service.go) call `secondaryNetworkAdd` (a private method of the IPAM driver, even though the call is possible because it's the same package), which then calls `SecondaryNetworkAllocate` (a public method).

Reply:
Changed to `SecondaryNetworkRelease(getAllocationPodOwner(args, k8sArgs, nil, true))`

For `secondaryNetworkAdd/secondaryNetworkDel`, earlier I hoped to keep `getAllocationPodOwner()` and allocation owner code in antrea_ipam.go, not in ipam_service.go. But sure, you can argue now with `SecondaryNetworkAllocate/SecondaryNetworkRelease`, the external callers already construct `PodOwner`. I slightly prefer to keep the current way (`getAllocationPodOwner()` in antrea_ipam.go), but no strong opinion. Let me know what you would suggest.